dns.go

```diff
@@ -134,9 +134,11 @@ func (rr *RFC3597) ToRFC3597(r RR) error {
 
 // fromRFC3597 converts an unknown RR representation from RFC 3597 to the known RR type.
 func (rr *RFC3597) fromRFC3597(r RR) error {
-	*r.Header() = rr.Hdr
+	hdr := r.Header()
+	*hdr = rr.Hdr
+	hdr.Rdlength = uint16(len(rr.Rdata) / 2)
```

Comment:
this is the crucial line. The other changes might be helping, but backing them out and focusing on the minimal fix it seems to boil down to:
~~~
*r.Header() = rr.Hdr
r.Header().Rdlength = uint16(len(rr.Rdata) / 2)
~~~
which probably also requires a comment about the /2 along the lines of 2 hex digits for every octet (or some such)

Reply:
The changes to `(*RFC3597).parse` are also important, otherwise it can overflow a `uint16`. I've updated it to be a bit clearer. It should be minimal enough now.